python/GafferSceneUI/RenderPassEditor.py

```diff
@@ -768,15 +769,25 @@ def __renderPassNames( self, plug ) :
 
 	def __addButtonClicked( self, button ) :
 
-		editScopeInput = self.__settingsNode["editScope"].getInput()
-		if editScopeInput is None :
-			return
+		menuDefinition = IECore.MenuDefinition()
+		self.addRenderPassButtonMenuSignal()( menuDefinition, self )
 
-		editScope = editScopeInput.node()
-		dialogue = _RenderPassCreationDialogue( self.__renderPassNames( self.__settingsNode["in"] ), editScope )
-		renderPassName = dialogue.waitForText( parentWindow = button.ancestor( GafferUI.Window ) )
-		if renderPassName :
-			self.__addRenderPass( renderPassName, editScope )
+		if menuDefinition.size() == 0 :
+			editScopeInput = self.__settingsNode["editScope"].getInput()
+			if editScopeInput is None :
+				return
+
+			editScope = editScopeInput.node()
+			dialogue = _RenderPassCreationDialogue( self.__renderPassNames( self.__settingsNode["in"] ), editScope )
+			renderPassName = dialogue.waitForText( parentWindow = button.ancestor( GafferUI.Window ) )
+			if renderPassName :
+				self.__addRenderPass( renderPassName, editScope )
+		elif menuDefinition.size() == 1 :
+			_, item = menuDefinition.items()[0]
+			item.command()
+		else :
+			self.__popupMenu = GafferUI.Menu( menuDefinition )
```

Comment:
Should we be adding a generic "Add" option here, so that the default functionality is always available?

Reply:
Included the generic "Add..." menu item in 8548823.